pkg/networkservice/common/roundrobin/server.go

```diff
@@ -50,19 +50,21 @@ func (s *selectEndpointServer) Request(ctx context.Context, request *networkserv
 	candidates := discover.Candidates(ctx)
 
 	for i := 0; i < len(candidates.Endpoints); i++ {
-		endpoint := s.selector.selectEndpoint(candidates.NetworkService, candidates.Endpoints)
-		if endpoint == nil {
-			return nil, errors.Errorf("failed to find endpoint for Network Service: %v %v", candidates.NetworkService, candidates.Endpoints)
-		}
-		u, err := url.Parse(endpoint.Url)
-		if err != nil {
-			return nil, errors.WithStack(err)
-		}
-		ctx = clienturlctx.WithClientURL(ctx, u)
-		request.GetConnection().NetworkServiceEndpointName = endpoint.Name
-		resp, err := next.Server(ctx).Request(ctx, request)
-		if err == nil {
-			return resp, err
+		for j := 0; j < len(candidates.Endpoints[i]); j++ {
+			endpoint := s.selector.selectEndpoint(candidates.NetworkService, candidates.Endpoints[i])
```

Comment:
I'm not sure how critical this is, but this will now work correctly.
`s.selector` holds an index of endpoint for this network service. This was _probably_ fine when we had an array of candidates for NS, but now we will have a shared index for all of the sub-arrays.
It would probably be better to modify the selector to handle double array instead of making double loop here.

Reply:
Though, I didn't research this, but I feel like the selector already had potential issues when we had network services defined with matches.